pyrealm/splash/evap.py

```diff
@@ -0,0 +1,316 @@
+    tc: InitVar[np.ndarray]
+    kWm: np.ndarray = np.array([150.0])
+
+    sat: np.ndarray = field(init=False)
+    """Slope of saturation vapour pressure temperature curve, Pa/K"""
+    lv: np.ndarray = field(init=False)
+    """Enthalpy of vaporization, J/kg"""
+    pw: np.ndarray = field(init=False)
+    """Density of water, kg/m^3"""
+    psy: np.ndarray = field(init=False)
+    """Psychrometric constant, Pa/K"""
+    econ: np.ndarray = field(init=False)
+    """Water-to-energy conversion factor"""
+    cond: np.ndarray = field(init=False)
+    """Daily condensation, mm"""
+    eet_d: np.ndarray = field(init=False)
+    """Daily EET, mm"""
+    pet_d: np.ndarray = field(init=False)
+    """Daily PET, mm"""
+    rx: np.ndarray = field(init=False)
+    """Variable substitute, (mm/hr)/(W/m^2)"""
+
+    def __post_init__(self, pa: np.ndarray, tc: np.ndarray) -> None:
+        """Calculate invariant components of evapotranspiration.
+
+        The post_init method calculates the invariant components of the
+        evapotranspiration fluxes. Two remaining components, the intersection hour angle
+        (hi), degrees and the estimated daily AET (aet_d), mm, depend on the estimated
+        evaporative supply rate, which can be updated during spin up the SPLASH model.
+        """
+
+        # Slope of saturation vap press temp curve, Pa/K
+        self.sat = sat_slope(tc)
+
+        # Enthalpy of vaporization, J/kg
+        self.lv = enthalpy_vap(tc)
+
+        # Density of water, kg/m^3
+        self.pw = density_h2o(tc, pa)
+
+        # Psychrometric constant, Pa/K
+        self.psy = psychro(tc, pa)
+
+        # Calculate water-to-energy conversion (econ), m^3/J
+        self.econ = self.sat / (self.lv * self.pw * (self.sat + self.psy))
+
+        # Calculate daily condensation (cn), mm
+        self.cond = (1e3) * self.econ * np.abs(self.solar.rnn_d)
+
+        # Estimate daily EET (eet_d), mm
+        self.eet_d = (1e3) * self.econ * self.solar.rn_d
+
+        # Estimate daily PET (pet_d), mm
+        self.pet_d = (1.0 + kw) * self.eet_d
+
+        # Calculate variable substitute (rx), (mm/hr)/(W/m^2)
+        self.rx = (3.6e6) * (1.0 + kw) * self.econ
+
+    def estimate_aet(
+        self, wn: np.ndarray, day_idx: Optional[int] = None, only_aet: bool = True
+    ) -> Union[np.ndarray, tuple[np.ndarray, np.ndarray, np.ndarray]]:
+        """Estimate actual evapotranspiration.
+
+        This method estimates the estimated daily actual evapotranspiration (AET,
+        mm/day), given estimates of the soil moisture  (wn) for observations.
+        Optionally, the method can also return the the intersection hour angle (hi,
+        degrees) and evaporative supply rate (sw, mm/h).
+
+        By default, sw is expected to provide estimates for all observations across all
+        days in the model, but day_idx can be set to provide sw for only one particular
+        day of observations.
+
+        Args:
+            sw: The soil moisture (mm).
```

Comment:
wn?

Reply:
Yeah - the old implementation passed `sw` in to this function (`calc_one_day`) and it was only ever used inside the function. So I standardised on passing `wn` around but forgot to update these docs.